docs/src/guides/getting-started.md

```diff
@@ -0,0 +1,169 @@
+with dream2nix and its [module system ](../modules.md).
+
+## Start a project
+
+We start by creating a new git repository with the following two files:
+
+- `flake.nix` declares our inputs, dream2nix and nixpkgs, as well as a single package `hello` as an output.
+  The package is declared by calling `dream2nix.lib.evalModules` with the definitions in `hello.nix`, nixpkgs
+  and a helper module to let dream2nix know about your directory layout.
+
+- `hello.nix` declares a dream2nix module that imports [mkDerivation](../reference/mkDerivation/index.md) and
+  uses that to build GNU hello.
+
+Check out the code below and its annotations to learn more!
+
+!!! note
+
+    And do not hesitate to message us on [matrix: #dream2nix:nixos.org](https://matrix.to/#/#dream2nix:nixos.org) if you found
+    a mistake or if things are unclear!
+
+### `flake.nix`
+
+```nix title="flake.nix"
+{
+  description = "A flake for my dream2nix packages";
+
+  inputs = { # (1)
+    dream2nix.url = "github:nix-community/dream2nix";
+    nixpkgs.follows = "dream2nix/nixpkgs";
+  };
+
+  outputs = {
+    self,
+    dream2nix,
+    nixpkgs,
+  }:
+  let
+    eachSystem = nixpkgs.lib.genAttrs [ # (2)
+      "aarch64-darwin"
+      "aarch64-linux"
+      "x86_64-darwin"
+      "x86_64-linux"
+    ];
+  in {
+    packages = eachSystem (system: {
+      hello = dream2nix.lib.evalModules { # (3)
+        packageSets.nixpkgs = nixpkgs.legacyPackages.${system}; # (4)
+        modules = [
+          ./hello.nix # (5)
+          { # (6)
+            paths.projectRoot = ./.;
+            paths.projectRootFile = "flake.nix";
+            paths.package = ./.;
+          }
+        ];
+      };
+      default = self.packages.${system}.hello;  # (7)
+    });
+  };
+}
+```
+
+1. Import dream2nix and tell nix to use whatever version of nixpkgs dream2nix declares. You can use other versions, but this it what we run our automated tests with.
+2. Define a helper function that allows us to reduce boilerplate and still support all of of the listed systems for our package.
+3. Create our package, called `hello` here, by *evaluating* the given dream2nix [modules](../modules.md).
+4. Pass the given instance of nixpkgs to build our package with as a *module argument*.
+5. Include our package definition from `hello.nix`. See below for its contents!
+6. Define relative paths to aid dream2nix to find lock files and so on during *evaluation time*. These settings should work for repos containing multiple python projects as simpler ones.
+7. We declare `hello` to be the default package. This allows us to just call i.e. `nix build .#` instead of `nix build .#hello`.
+
+### `hello.nix`
+
+```nix title="hello.nix"
+{ # (1)
+  dream2nix,
+  config,
+  lib,
+  ...
+}: {
+  imports = [
+    dream2nix.modules.dream2nix.mkDerivation # (2)
+  ];
+
+  # (3)
+  name = "hello";
+  version = "2.12.1";
+
+  # (4)
+  # deps = {nixpkgs, ...}: {
+  #   inherit
+  #    (nixpkgs)
+  #    stdenv
+  #    ;
+  # };
```

Comment:
Maybe remove this for now? The `deps` concept could still be introduced in another guide.
Also I'm not even sure if we should keep `deps`, or if we better just provide access to `pkgs`. The user can still use overlays to override deps etc.

Reply:
Yeah, same applies for the lock mention. I'd suggest moving it into a modules guide when thats done :)